src/main/java/seedu/address/logic/commands/OweCommand.java

```diff
@@ -0,0 +1,136 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOUR;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.student.Address;
+import seedu.address.model.student.Email;
+import seedu.address.model.student.Name;
+import seedu.address.model.student.OwedAmount;
+import seedu.address.model.student.PaidAmount;
+import seedu.address.model.student.Phone;
+import seedu.address.model.student.Rate;
+import seedu.address.model.student.Schedule;
+import seedu.address.model.student.Student;
+import seedu.address.model.student.Subject;
+
+/**
+ * Updates the amount of tuition fee owed by an existing student in the address book.
+ */
+public class OweCommand extends Command {
+
+    public static final String COMMAND_WORD = "owe";
+    public static final String COMMAND_WORD_RANDOM_CASE = "owE";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Updates the amount the student owes "
+            + "by the index number used in the displayed Student list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_HOUR + "HOUR OWED (must be a positive multiple of 0.5)\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_HOUR + "2 ";
+
+    public static final String MESSAGE_UPDATE_OWED_AMOUNT_SUCCESS = "Updated Student owedAmount: %1$s";
+    public static final String MESSAGE_MISSING_HOUR = "Number of hours owed must be provided.";
+
+    private final Index index;
+    private final double hourOwed;
+
+    /**
+     * @param index of the student in the filtered student list to update owedAmount.
+     * @param hourOwed number of hours the student owes.
+     */
+    public OweCommand(Index index, double hourOwed) {
+        this.index = index;
+        this.hourOwed = hourOwed;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Student> lastShownList = model.getFilteredStudentList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        Student studentToEdit = lastShownList.get(index.getZeroBased());
+        Student editedStudent = createEditedStudent(studentToEdit, hourOwed);
+
+        model.setStudent(studentToEdit, editedStudent);
+        model.updateFilteredStudentList(Model.PREDICATE_SHOW_ALL_STUDENTS);
+
+        return new CommandResult(String.format(MESSAGE_UPDATE_OWED_AMOUNT_SUCCESS, Messages.format(editedStudent)));
```

Comment:
Could consider standardising the output in the output box for both pay and owe command